src/lib/validate-input.js

```diff
@@ -623,6 +623,42 @@ export function validateGatherParams(input, indices, {axis = 0} = {}) {
   }
 }
 
+export function validateGatherNDParams(input, indices) {
+  const inputRank = input.rank;
+  const inputShape = input.shape;
+  const indicesRank = indices.rank;
+  const indicesShape = indices.shape;
+
+  // Refer to https://docs.openvino.ai/2024/documentation/openvino-ir-format/operation-sets/ +
+  //   operation-specs/movement/gather-nd-8.html
```

Comment:
Why not refer to WebNN spec?

Reply:
Current WebNN spec doesn't define this gatherND operator, so I refer to this resources.